tests/custom/flatten_lex/sets/set_of_int/set.essence

```diff
@@ -0,0 +1,3 @@
+find A : set of int(1..10) 
+find B : set of int(1..10) 
+such that A .< B
```

Comment:
Should we be testing for this in the input? I am sure we don't support .< between two different representations. You get away with it here thanks to compact.

Just do `find s : set (size 2) of set of int(1..10)` which would create .< constraints internally then refine them?

Reply:
Looking at the rest of the files now I see there is a bunch of other .< constraints in the input.

Maybe we can group them in a `tests/custom/dotless` directory? I think these will eventually go and be replaced by just `<`.

Reply:
I thought the plan was to get .< working in the way we want < to be? Or should it only be supporting the core tree language?

There are lots of tests with .< I think. They will of course be changed to < when we upgrade its power.

Reply:
We want to leave .< alone. It works by treating all of the variables of the representation as a block and putting lex between them. The two operands must have the same domain and must be represented in the same way.

We also want to leave ~< alone as well. This is a correct way of ordering anything independent of representation or domain. They must have the same type of course.

We want to implement < for all types using symmetryOrder etc. The hope is this new family of comparators will be as efficient as .< and as generic as ~<.

Reply:
Okay. Currently I match on .< and flatten both sides. So I should instead do that with <?

Reply:
What do you mean by leave .< alone?

> We want to implement < for all types using symmetryOrder etc. The hope is this new family of comparators will be as efficient as .< and as generic as ~<.

This is currently done for .< instead of <

https://github.com/conjure-cp/conjure/blob/de0550be1868836a9dab6963f9a8adbc2bf9f3a0/src/Conjure/UI/Model.hs#L1964-L1976

Reply:
After rule_DotLtLeq we match on <lex and in cases where each side is not a flat matrix/list we flatten. Are you suggesting a different mechanism for this?

Reply:
I've been doing some archeology and it seems this function has changed somewhat since you first implemented it.

https://github.com/conjure-cp/conjure/blob/2c7144e970047e7b69695f5bf322b9d0aee50359/src/Conjure/UI/Model.hs#L1724-L1786

Should we make sure we are still conforming to the .< and ~< specs before moving forward? I'm not sure we have sufficient testing on those to be sure they haven't been broken in our quest to implement the fully generic comparator.

Reply:
Perhaps at this point we should merge this pull request since it is a fairly minimal fix for #456? We can then spin up a separate issue that aims to straighten up our comparators and make sure they conform to spec.

Reply:
What about the comment lines in the tests?

I am really not sure about the value of testing .< from the input since it has a very specific internal use case. It is generated by the representation rules, and refined away later.

Reply:
I will push the comment lines filter onto this PR.

Are we okay with testing .< on the treelike structures?

```
given A : matrix indexed by [int(0..1)] of (bool,bool)
find B : matrix indexed by [int(0..1)] of (bool,bool)
such that A .< B
```

Reply:
Let's keep them in if you are keen. My main point is we shouldn't be
touching .< or ~< at all anyway.

Reply:
Okay, I just want to conform to the spec. What do you mean by touching .< at all? 

Do you want to revert to rule_DotLtLeq before it called symmetryOrdering and introduce a rule_LtLeq that does what rule_DotLtLeq currently does?

Reply:
I mean, AFAIK, before this PR the dot-less handling is correct. And we don't want to change it. We don't necessarily want to test it in this way either since it is generated as part of almost all tests and works correctly anyway.

Am I missing something?

And yes, the new implementation that will work on everything should just work on <.

Reply:
Okay, so if I remove the test on sets which doesn't make sense for .< and clean up the comments in the expected files we should be good to merge?

I'll work on the < implementation in a separate branch from this.

Reply:
Sounds good to me! I am at work today so should be able to stay in the loop as you commit more stuff to this pr.

Reply:
Done!